gradio/routes.py

```diff
@@ -376,14 +364,17 @@ async def conditional_routing_middleware(request: Request, call_next):
                     and path not in ["/config", "/login"]
                     and not path.startswith("/theme")
                 ):
-                    print("proxying")
-                    try:
+                    if App.app_port is None:
+                        App.app_port = request.url.port or int(
+                            os.getenv("GRADIO_SERVER_PORT", "7860")
+                        )
```

Comment:
The reason for this is that the port isn't actually present on every request but it always is for the first. SO we grab it when we have it and reuse, it will never change.

Reply:
Interesting

Reply:
or infuriating, you can decide :D